core/src/main/java/org/elasticsearch/script/StoredScriptSource.java

```diff
@@ -266,8 +262,7 @@ public static StoredScriptSource parse(String lang, BytesReference content, XCon
                         //this is really for search templates, that need to be converted to json format
                         try (XContentBuilder builder = XContentFactory.jsonBuilder()) {
                             builder.copyCurrentStructure(parser);
-                            return new StoredScriptSource(lang, builder.string(),
-                                Collections.singletonMap(Script.CONTENT_TYPE_OPTION, XContentType.JSON.mediaType()));
+                            return new StoredScriptSource(lang, builder.string(), Collections.emptyMap());
```

Comment:
Since the check is being removed, maybe it makes sense to continue to store this with the content-type set to JSON?  This works now because of the default in the mustache engine.